packages/ldap-integration/src/helpers.ts

```diff
@@ -0,0 +1,43 @@
+import { createClient } from 'ldapjs';
+
+
+class UserGroupApiHelper {
+  private static UserGroupHelperInstance: UserGroupApiHelper;
+  ldapHost: string | any = process.env.LDAP_HOST;
+  ldapBase: string | any = process.env.LDAP_BASE;
+  constructor() { }
+  public static getAPIInstance() {
+    if (!UserGroupApiHelper.UserGroupHelperInstance) {
+      UserGroupApiHelper.UserGroupHelperInstance = new UserGroupApiHelper();
+    }
+    return UserGroupApiHelper.UserGroupHelperInstance;
+  }
+
+  // Helper function to fetch user/group profile from LDAP
+  public getProfilesBy(profile_param: string) {
+    return new Promise((resolve, reject) => {
+      const ldapClient = createClient({ url: this.ldapHost });
+      const search_options: Object = {
+        scope: `sub`,
+        filter: `${profile_param}`,
+        attributes: `*`
+      };
+      let profile: LdapType;
+      ldapClient.search(this.ldapBase, search_options, (err, response) => {
+        response.on(`searchEntry`, (entry) => {
+          console.log(`LDAP data found!!`);
+          profile = entry.object;
+        });
+        response.on(`error`, (error: Error) => {
+          console.error(`LDAP error: ` + error.message);
+        });
+        response.on(`end`, (result: any) => {
+          console.log(`LDAP status: ` + result.status);
```

Comment:
Again, do we need to log this to console. I think there can be a better way to do this.